src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -118,70 +98,69 @@ protected override bool ImprovementRequirementSatisfied(PeerInfo? bestPeer)
             bool shouldMoveToMain = (options & DownloaderOptions.MoveToMain) == DownloaderOptions.MoveToMain;
 
             int blocksSynced = 0;
-            long currentNumber = Math.Max(0, Math.Min(_blockTree.BestKnownNumber, bestPeer.HeadNumber - 1));
-            if (_logger.IsTrace) _logger.Trace($"MergeBlockDownloader GetCurrentNumber: currentNumber {currentNumber}, beaconPivotExists: {_beaconPivot.BeaconPivotExists()}, BestSuggestedBody: {_blockTree.BestSuggestedBody.Number}, BestKnownNumber: {_blockTree.BestKnownNumber}");
-        bool HasMoreToSync()
-                => currentNumber <= bestPeer!.HeadNumber;
-            while(ImprovementRequirementSatisfied(bestPeer!) && HasMoreToSync())
+            long currentNumber = _blockTree.BestKnownNumber;
+            if (_logger.IsTrace)
+                _logger.Trace(
+                    $"MergeBlockDownloader GetCurrentNumber: currentNumber {currentNumber}, beaconPivotExists: {_beaconPivot.BeaconPivotExists()}, BestSuggestedBody: {_blockTree.BestSuggestedBody.Number}, BestKnownNumber: {_blockTree.BestKnownNumber}, BestPeer: {bestPeer}, BestKnownBeaconNumber {_blockTree.BestKnownBeaconNumber}");
+
+            bool HasMoreToSync()
+                => _beaconPivot.BeaconPivotExists() && currentNumber < _blockTree.BestKnownBeaconNumber &&
+                   bestPeer.HeadNumber > _blockTree.BestKnownNumber;
+
```

Comment:
how does forward sync process blocks after beacon pivot? it seems like it doesn't check if it is processing after pivot